java.hints/src/org/netbeans/modules/java/hints/jdk/ConvertVarToExplicitType.java

```diff
@@ -119,15 +148,21 @@ protected void performRewrite(TransformationContext tc) throws Exception {
     private static boolean isLocalVarType(HintContext ctx) {
 
         CompilationInfo info = ctx.getInfo();
-        
+
         if (info.getSourceVersion().compareTo(SourceVersion.RELEASE_9) < 1) {
             return false;
-        }        
+        }
 
         TreePath treePath = ctx.getPath();
-
+        EnhancedForLoopTree efl = null;
+        TypeMirror expressionType = null;
+        if (treePath.getLeaf() instanceof EnhancedForLoopTree) {
```

Comment:
Thanks for the review, Arunava. Is there any advantage to using == operator instead of instanceof ?
As far as I am aware, EnhancedForLoopTree class is not subclassed anywhere currently and even if it is subclassed in the future, wouldn't we want var hint to be available there?

Reply:
It is advisable to use Tree kind check wherever possible instead of instanceof check, former is less costly.
But if you are expecting a new subclass whose kind might be different from parent tree, then instanceof check is the only option.
In that case please change below code at ConvertToVarHint.java:141  
'else if (statementPath.getLeaf().getKind() == Tree.Kind.ENHANCED_FOR_LOOP)'

Reply:
In fact an implementation class (i.e. a proxy, or some kind of adapter) could implement a number of interfaces, which can confuse `instanceof` checks. As a general rule, if an API defines an enum / tag whose values correspond to interfaces, the tag should be used in preference to `instanceof`, just to avoid possible future problems. It's true that `instanceof interface-type` costs more cycles than `instanceof class-type`.

Reply:
I understand. It does make more sense to change instanceof to equality check.